From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Wed, 3 Jun 2009 12:34:08 +0000 (+0400)
Subject: pidns: pi-futex pid check fixup
X-Git-Url: http://git.openvz.org/?p=linux-2.6.27-openvz;a=commitdiff_plain;h=81e645095ced04ab9bff9680f112774c396ab13f

pidns: pi-futex pid check fixup

fix WARN_ON condition

port rh5 patch from Stanichenko Marat <mstanichenko@openvz.org>

http://bugzilla.openvz.org/show_bug.cgi?id=1262
---

diff --git a/kernel/futex.c b/kernel/futex.c
index 1cc3092..d1351bb 100644
--- a/kernel/futex.c
+++ b/kernel/futex.c
@@ -534,7 +534,7 @@ lookup_pi_state(u32 uval, struct futex_hash_bucket *hb,
 
 			WARN_ON(!atomic_read(&pi_state->refcount));
 			WARN_ON(pid && pi_state->owner &&
-				pi_state->owner->pid != pid);
+				task_pid_vnr(pi_state->owner) != pid);
 
 			atomic_inc(&pi_state->refcount);
 			*ps = pi_state;
